Classes/ContentObject/Relation.php

```diff
@@ -122,37 +122,59 @@ public function cObjGetSingleExt(
     protected function getRelatedItems(
         ContentObjectRenderer $parentContentObject
     ) {
-        $relatedItems = [];
 
         list($localTableName, $localRecordUid) = explode(':',
             $parentContentObject->currentRecord);
 
         $localTableNameOrg = $localTableName;
+        $localFieldName = $this->configuration['localField'];
 
-        // pages has a special overlay table constriction
-        // @todo this check can be removed when TYPO3 8 support is dropped since pages translations are in pages then as well
-        if ($GLOBALS['TSFE']->sys_language_uid > 0 && $localTableName === 'pages' && Util::getIsTYPO3VersionBelow9()) {
-            $localTableName = 'pages_language_overlay';
+        if (!$this->isTcaSetForTablesField($localTableName, $localFieldName)) {
+            return [];
         }
 
-        $localTableTca = $GLOBALS['TCA'][$localTableName];
-        $localFieldName = $this->configuration['localField'];
+        $this->usePagesLanguageOverlayInsteadOfPagesIfNeeded($localFieldName);
 
-        if (isset($localTableTca['columns'][$localFieldName])) {
-            $localFieldTca = $localTableTca['columns'][$localFieldName];
-            $localRecordUid = $this->getUidOfRecordOverlay($localTableNameOrg, $localRecordUid);
-            if (isset($localFieldTca['config']['MM']) && trim($localFieldTca['config']['MM']) !== '') {
-                $relatedItems = $this->getRelatedItemsFromMMTable($localTableName,
-                    $localRecordUid, $localFieldTca);
-            } else {
-                $relatedItems = $this->getRelatedItemsFromForeignTable($localTableName,
-                    $localRecordUid, $localFieldTca, $parentContentObject);
-            }
+        $localFieldTca = $GLOBALS['TCA'][$localTableName]['columns'][$localFieldName];
+        $localRecordUid = $this->getUidOfRecordOverlay($localTableNameOrg, $localRecordUid);
+        if (isset($localFieldTca['config']['MM']) && trim($localFieldTca['config']['MM']) !== '') {
+            $relatedItems = $this->getRelatedItemsFromMMTable($localTableName,
+                $localRecordUid, $localFieldTca);
+        } else {
+            $relatedItems = $this->getRelatedItemsFromForeignTable($localTableName,
+                $localRecordUid, $localFieldTca, $parentContentObject);
         }
 
         return $relatedItems;
     }
 
+    /**
+     * @param string $localTableName
+     * @todo this can be removed when TYPO3 8 support is dropped since pages translations are in pages then as well
+     */
+    protected function usePagesLanguageOverlayInsteadOfPagesIfNeeded(string &$localTableName)
```

Comment:
Why do you pass a reference here for a string instead of returning the valu and reassign it in the calling method?